### PR TITLE
Sync practiced concepts for practice exercises

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/models/exercise/practiced_concept.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Exercise::PracticedConcept < ApplicationRecord
```
Comment (ErikSchierboom) [inline app/models/exercise/practiced_concept.rb:1]: Let me know what you think of the class name.

File: app/models/practice_exercise.rb [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+    class_name: "Exercise::PracticedConcept",
+    foreign_key: :exercise_id,
+    inverse_of: :exercise,
+    dependent: :destroy
+
+  has_many :practices,
+    through: :exercise_practiced_concepts,
+    source: :concept
+
+  def self.with_prerequisite(concept)
```
Comment (ErikSchierboom) [inline app/models/practice_exercise.rb:12]: I added this method, as the existing `that_practice` method was slightly confusing now that we actually have practice concepts.

File: app/models/practice_exercise.rb [hunk 3]
```diff
@@ -1,6 +1,21 @@
 class PracticeExercise < Exercise
-  def self.that_practice(concept)
+  has_many :exercise_practiced_concepts,
+    class_name: "Exercise::PracticedConcept",
+    foreign_key: :exercise_id,
+    inverse_of: :exercise,
+    dependent: :destroy
+
+  has_many :practices,
```
Comment (iHiD) [inline app/models/practice_exercise.rb]:
```suggestion
  has_many :practiced_concepts,
```

File: app/models/exercise/practiced_concept.rb (hunk 1 above)
Comment (iHiD) [reply app/models/exercise/practiced_concept.rb:1]: Yep - Like it!

**Pull request merged**